[Server] Allow overriding the default file name pattern for Discovery (#292)
modelcontextprotocol/php-sdk: docs/server-builder.md

    ->setDiscovery(
        basePath: __DIR__,
        scanDirs: ['.', 'src', 'lib'],           // Where to look for MCP attributes
        excludeDirs: ['vendor', 'tests'],        // Where NOT to look
        cache: $cacheInstance,                   // Optional: cache discovered elements

>         namePatterns: ['*.php', '*.inc'],         // Optional: list of filename patterns to match

Reviewer: looks like a small whitespace inconsistency

```suggestion
        excludeDirs: ['vendor', 'tests'],        // Where NOT to look
        cache: $cacheInstance,                   // Optional: cache discovered elements
        namePatterns: ['*.php', '*.inc'],        // Optional: list of filename patterns to match
```

- `$scanDirs` (array): Directories to recursively scan for `#[McpTool]`, `#[McpResource]`, etc. All subdirectories are included. (default: `['.', 'src']`)
- `$excludeDirs` (array): Directory names to exclude **within** the scanned directories during recursive scanning
- `$cache` (CacheInterface|null): Optional PSR-16 cache to store discovered elements for performance

> - `$namePatterns` (array): Optional list of Finder->name() compatible patterns to match against file names (default: `['*.php']`)

Reviewer: Exposing `Finder->name()` here feels a bit internal to me, let's focus on the behavior:
```suggestion
- `$namePatterns` (array): Optional list of patterns (regexp, glob, or string) for file names (default: `['*.php']`)
```
